examples/flux/math.py

```diff
@@ -0,0 +1,38 @@
+import math
+from typing import Tuple
+
+import mindspore as ms
+from mindspore import Tensor, ops
+
+
+def attention(q: Tensor, k: Tensor, v: Tensor, pe: Tensor) -> Tensor:
+    q, k = apply_rope(q, k, pe)
+
+    # x = torch.nn.functional.scaled_dot_product_attention(q, k, v)
+    # x = rearrange(x, "B H L D -> B L (H D)")
+    scale_factor = float(1 / math.sqrt(q.shape[-1]))
+    x = ops.operations.nn_ops.FlashAttentionScore(head_num=q.shape[1], input_layout="BNSD", scale_value=scale_factor)(
+        q.to(ms.bfloat16), k.to(ms.bfloat16), v.to(ms.bfloat16), None, None, None, None
```

Comment:
why manually cast q k v to bf16? which is not supported in PyNative

Reply:
Because flashattention op only supported fp16/bf16, in this case we could remove this manual casting as all tensors inputs are bf16 already, but IMO for generality we should keep it and it works in ms2.3